### PR TITLE
feat: slicing order page and prepare it for integration

### PR DESCRIPTION
**Pull request by mafiefa02**
# Note
I improvised a bit with the order item card design. It's just using `Card` component is general, where the vendor name is using the card title. Comparison attached (left: what I did, right: figma design)
image

# Details
1. Add review form is in the shape of `{ rating: number, review: string }`
2. See invoice modal is currently using a placeholder
3. Upload receipt form is in the shape of `{ receipt: string (url to the photo) }`
4. Default tab is `ordered` tab.
5. Temporary interfaces/types is in the file `orderInterface.ts` in shared models
6. Modal is place within every tab component (ie. `NeedToPayTab.tsx` etc.), but it's dynamically imported so should be fine (cmiiw), this is to avoid having the entire page as a client component.

# UI Preview
### Tab Loading State
image

### Ordered Tab
image

### Need to Pay Tab
image

### Paid Tab
image

### Finished Tab
image

### See Billing Page
image

### Upload Receipt Modal
image

### Add Review Modal
image

### CODE REVIEW

File: src/app/order/(default)/container/OrderTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use client';
```
Comment (galihpermana29) [inline src/app/order/(default)/container/OrderTabs.tsx]: Put this file inside the Tab folder

File: src/app/order/(default)/container/OrderTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return router.push(`${pathname}?${tabParam}`);
+  }
+
+  const activeKeyIsValid = activeKey
+    ? tabs.map((tab) => tab.key).includes(activeKey)
+    : false;
+
+  useEffect(() => {
+    if (!activeKeyIsValid) return handleChangeTab('ordered');
+  }, []);
```
Comment (galihpermana29) [inline src/app/order/(default)/container/OrderTabs.tsx]: Actually you can put this file inside the custom hooks that you created named useGetTabToDisplay. return the function and required variable. Call the hooks inside of this file.

Reason: not clean a clean code and hard to understand

File: src/app/order/(default)/page.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    activeTab,
+    tabs,
+  });
+
+  return (
+    <Suspense fallback={<TabLoading withAlert />}>
+      {tabToDisplay.key === 'ordered' && <OrderedTab />}
+      {tabToDisplay.key === 'need-to-pay' && <NeedToPayTab />}
+      {tabToDisplay.key === 'paid' && <PaidTab />}
+      {tabToDisplay.key === 'finished' && <FinishedTab />}
```
Comment (galihpermana29) [inline src/app/order/(default)/page.tsx]: Refactor this with object

const tabSection = {
  odered: the component
}

return (
suspense...
{tabSection[tabToDisplay.key]}
)

File: src/app/order/(default)/container/Tab/Finished/FinishedTab.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
+'use client';
```
Comment (galihpermana29) [inline src/app/order/(default)/container/Tab/Finished/FinishedTab.tsx]: rename the folder to FinishTabSection, NeedToPayTabSection, and so on.

**Review by galihpermana29 — COMMENTED** [review]
Overall really good code. Love this, please solve the feedbacks and merge

File: src/app/order/(default)/container/OrderTabs.tsx (hunk 2 above)
Comment (mafiefa02) [reply src/app/order/(default)/container/OrderTabs.tsx]: I did this a bit different from what you suggested. I need to use the `useGetTabToDisplay` hook on server so I can't really move the `handleChangeTab` function into the same file. I ended up still doing the `activeKeyIsValid` inside `OrderTabs.tsx` and moved the `handleChangeTab` function to a custom hook.

Might need another review @galihpermana29

**Pull request merged**